internal/preflight/preflight.go

```diff
@@ -195,3 +195,36 @@ func (p *preflightHelper) verifySign(ctx context.Context, pv *kmmv1beta1.Preflig
 	}
 	return false, "Waiting for sign verification"
 }
+
+func DeleteModuleStatus(statuses *[]v1beta2.PreflightValidationModuleStatus, nsn types.NamespacedName) {
+	*statuses = slices.DeleteFunc(*statuses, func(status v1beta2.PreflightValidationModuleStatus) bool {
+		return status.Namespace == nsn.Namespace && status.Name == nsn.Name
+	})
+}
+
+func FindModuleStatus(statuses []v1beta2.PreflightValidationModuleStatus, nsn types.NamespacedName) (*v1beta2.PreflightValidationModuleStatus, bool) {
+	i := slices.IndexFunc(statuses, func(status v1beta2.PreflightValidationModuleStatus) bool {
+		return status.Namespace == nsn.Namespace && status.Name == nsn.Name
+	})
+
+	if i != -1 {
+		return &statuses[i], true
+	}
+
+	return nil, false
+}
+
+func UpsertModuleStatus(statuses *[]v1beta2.PreflightValidationModuleStatus, s v1beta2.PreflightValidationModuleStatus) error {
```

Comment:
The same for this one: if it is only used in test, it should be defined in the test file

Reply:
It is only used in the tests, but in two different packages, so it cannot be in a _test.go file. Do you think it should go somewhere else?

Reply:
same answer as above